### ISSUE TITLE
Use `BossSceneController.IsBossScene` to detect boss scene

### ISSUE THREAD

**Issue by royitaqi** [royitaqi/HollowKnight.GodhomeWinLossTracker#7]
* [/] Use `BossSceneController.IsBossScene` to detect boss scene
    * [/] This works differently than the message bus way. This function checks the actual current scene, while my implementation checks the given scene name (no matter it's a current scene or not).
  * [x] Alternatively, aggregate boss scene name in `GodhomeUtils` and use this to detect boss vs non-boss, instead of relying on non-boss scene names, which isn't exhaustive when outside godhome and leads to the following problem.

**Comment by royitaqi** [comment]
Problem with non-exhaustive non-boss scene names:
```
[INFO]:[GodhomeWinLossTracker] - OnSceneLoad: Tutorial_01
[INFO]:[GodhomeWinLossTracker] - Message on bus: Scene changed to Tutorial_01
[ERROR]:[API] - GodhomeWinLossTracker.AssertionFailedException: Boss scene name Tutorial_01 should exist in BossSceneToName
[ERROR]:[API] -   at GodhomeWinLossTracker.DevUtils.Assert (System.Boolean condition, System.String message) [0x00011] in D:\GitHubRepositories\HollowKnight.GodhomeWinLossTracker\GodhomeWinLossTracker\DevUtils.cs:20 
[ERROR]:[API] -   at GodhomeWinLossTracker.GodhomeUtils.GetNullableBossName (System.String sceneName) [0x00010] in D:\GitHubRepositories\HollowKnight.GodhomeWinLossTracker\GodhomeWinLossTracker\GodhomeUtils.cs:27 
[ERROR]:[API] -   at GodhomeWinLossTracker.MessageBus.Handlers.BossChangeDetector.OnMessage (GodhomeWinLossTracker.MessageBus.TheMessageBus bus, Modding.Loggable logger, GodhomeWinLossTracker.MessageBus.IMessage msg) [0x0002a] in D:\GitHubRepositories\HollowKnight.GodhomeWinLossTracker\GodhomeWinLossTracker\MessageBus\Handlers\BossChangeDetector.cs:20 
[ERROR]:[API] -   at GodhomeWinLossTracker.MessageBus.TheMessageBus.Put (GodhomeWinLossTracker.MessageBus.IMessage message) [0x0004a] in D:\GitHubRepositories\HollowKnight.GodhomeWinLossTracker\GodhomeWinLossTracker\MessageBus\TheMessageBus.cs:48 
[ERROR]:[API] -   at GodhomeWinLossTracker.GodhomeWinLossTracker.OnSceneLoad (System.String sceneName) [0x00013] in D:\GitHubRepositories\HollowKnight.GodhomeWinLossTracker\GodhomeWinLossTracker\GodhomeWinLossTracker.cs:85 
[ERROR]:[API] -   at Modding.ModHooks.BeforeSceneLoad (System.String sceneName) [0x00030] in <2b54cdaf7a934f0eb297d50c4090ea9f>:0 
[INFO]:[CustomKnight] - skinable Geo : material is null
[INFO]:[CustomKnight] - [SwapManager] There are no folders in the Swap directory. No textures to Swap.
```

**Comment by royitaqi** [comment]
The alternative approach is implemented in ee351d5192976b786374719af974bf209b252d01
